src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs

```diff
@@ -184,6 +185,26 @@ public void EndPrint_SetValue_ReturnsExpected()
             }
         }
 
+        [ConditionalFact(nameof(CanPrintToPdf))]
+        public void Print_DefaultPrintController_Success()
+        {
+            bool endPrintCalled = false;
+            var endPrintHandler = new PrintEventHandler((sender, e) => endPrintCalled = true);
+            using (var document = new PrintDocument())
+            {
+                document.PrinterSettings.PrinterName = PrintToPdfPrinterName;
+                document.PrinterSettings.PrintFileName = GetTestFilePath();
+                document.PrinterSettings.PrintToFile = true;
+                document.EndPrint += endPrintHandler;
+                document.Print();
+                document.EndPrint -= endPrintHandler;
+            }
+
+            // File may not have finished saving to disk when Print returns,
```

Comment:
Maybe there's a way to wait for a file to finish writing.

Reply:
Waiting on disk I/O is notoriously unreliable. Testing printing I/O in general is borderline but coupling that with disk I/O seems like the flakiest test possible.

Reply:
I think this is dependent on how the printer handles things, so I didn't see a good way of dealing with it. I didn't want to do some sort of arbitrary timeout/polling and the main goal was to actually call the default printer controller and make sure it didn't error, not necessarily test the 'Microsoft Print to PDF' printer, so I figured making sure we hit `EndPrint` was a reasonable check.